lib/dialects/oracle/query-generator.js

```diff
@@ -0,0 +1,611 @@
+  },
+
+  dropTableQuery: function(tableName, options) {
+    options = options || {};
+
+    var query = [
+        'DECLARE',
+          'V_COUNT INTEGER;',
+          'V_CURSOR_NAME INTEGER;',
+          'V_RET INTEGER;',
+        'BEGIN',
+          'SELECT COUNT(1) INTO V_COUNT FROM ALL_TABLES WHERE TABLE_NAME = <%= tablestr %> AND OWNER = <%= ownerstr %>;',
+          'IF V_COUNT = 1 THEN',
+            'V_CURSOR_NAME := DBMS_SQL.OPEN_CURSOR;',
+            'DBMS_SQL.PARSE(V_CURSOR_NAME, \'DROP TABLE <%= table %> CASCADE CONSTRAINTS\', DBMS_SQL.NATIVE);',
+            'V_RET := DBMS_SQL.EXECUTE(V_CURSOR_NAME);',
+          'END IF;',
+        'END;',
+    ].join(' ');
+
+    var strTable = tableName;
+    var strOwner = 'USER';
+    if (tableName.tableName) {
+      strTable = tableName.tableName;
+      strOwner = wrapSingleQuote(tableName.schema);
+    }
+
+    return _.template(query)({
+      table: this.quoteTable(tableName),
+      tablestr: wrapSingleQuote(strTable),
+      ownerstr: strOwner
+    });
+  },
+
+  // Sends auto increment field to query-interface function. That means that it should do more queries to create trigger and sequence
+  // For 12.1 version you can return null and then change createTableQuery to not remove IDENTITY attribute for that version.
+  checkTableAutoinc: function(tableName, attributes) {
+    for (var attr in attributes) {
+      var dataType = attributes[attr];
+      if (Utils._.includes(dataType, 'GENERATED BY DEFAULT AS IDENTITY')) {
+        return attr;
+      }
+    }
+    return null;
+  },
+
+  createTableQuery: function(tableName, attributes, options) {
+    var query = 'CREATE TABLE <%= table %> (<%= attributes%>)'
+      , attrStr = []
+      , primaryKeys = [];
+
+    for (var attr in attributes) {
+      var dataType = attributes[attr];
+      if (Utils._.includes(dataType, 'PRIMARY KEY')) {
+        primaryKeys.push(attr);
+        dataType = dataType.replace(/ GENERATED BY DEFAULT AS IDENTITY/, '');
+        attrStr.push(this.quoteIdentifier(attr) + ' ' + dataType.replace(/ PRIMARY KEY/, ''));
+      } else {
+        attrStr.push(this.quoteIdentifier(attr) + ' ' + dataType);
+      }
+    }
+
+    var values = {
+      table: this.quoteTable(tableName),
+      attributes: attrStr.join(', '),
+    },
+    pkString = primaryKeys.map(function(pk) { return this.quoteIdentifier(pk); }.bind(this)).join(', ');
+    if (pkString.length > 0) {
+      values.attributes += ', PRIMARY KEY (' + pkString + ')';
+    }
+
+    var sql = _.template(query)(values).trim();
+    return this.replaceBooleanDefaults(sql);
+  },
+
+  booleanValue: function(value){
+    return !!value ? 1 : 0;
+  },
+
+  addWhereLimitAndOffset: function(options){
+    if (options.offset && !options.limit) {
+    } else if (options.limit) {
+      if (options.offset) {
+      } else {
+        options.where.ROWNUM = {$lte: options.limit};
+      }
+    }
+    return options;
+  },
+
+  addLimitAndOffset: function(options){
```

Comment:
Can't this function just return `''`? As I understand it this will never work in oracle?